text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,786 @@
+
+## Working Backward
+
+When upgrading CDK constructs, CDK users follow these steps using our construct upgrade guides and CDK CLI
+commands:
+
+- **Review Migration Guide:** Developers start by reading CDK-provided comprehensive migration guide for
+their specific construct upgrade (e.g., DynamoDB Table to TableV2). The details of individual migration
+guide is out of scope of this RFC.
+- **Update CDK Constructs:** Developers update their CDK stack to V2 variants of the constructs following the
+construct upgrade guide.
+- **Validate Changes:** Developers run CDK CLI commands to validate their code changes.
+- **Execute Changes:** Developers run the corresponding execute command in CDK CLI to deploy the changes.
+
+### Working Backward: DynamoDB Table Upgrade Example
+
+Below is a common upgrade scenario to demonstrate how the CDK Construct Upgrade framework helps developers
+safely migrate their infrastructure. Consider a DynamoDB table deployed to CloudFormation using the legacy
+`Table` construct:
+
+```ts
+import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';
+
+export class DemoStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const table = new dynamodb.Table(this, 'MyTable', {
+      partitionKey: {
+        name: 'PK',
+        type: dynamodb.AttributeType.STRING,
+      },
+      replicationRegions: ['us-west-2'],
+    });
+  }
+}
+```
+
+**Step 1:**: CDK users follow the construct upgrade guide provided by CDK team and upgrade to TableV2 construct:
+
+```ts
+import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';
+
+export class DemoStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const table = new dynamodb.TableV2(this, 'MyTable', {
+      partitionKey: {
+        name: 'PK',
+        type: dynamodb.AttributeType.STRING,
+      },
+      tableName: 'DemoStack-MyTable794EDED1-11W4MR8VZ0UPE',
+      replicas: [
+        {
+          region: 'us-west-2',
+        }
+      ]
+    });
+  }
+}
+```
+
+**Step 2:**: CDK users follow the construct upgrade guide to validate code changes by running the
+CDK CLI commands below:
+
+```s
+# exit with 1 if CloudFormation drift is detected for resource type AWS::DynamoDB::Table
+# exit with 0 otherwise
+$ cdk drift --rejectAWS::DynamoDB::Table
+
+Stack DemoStack
+✅ No CloudFormation drift detected
+
+# exit with 1 if resources types AWS::DynamoDB::Table and Custom::DynamoDBReplica are deleted, OR
+# exit with 1 if resources type AWS::DynamoDB::GlobalTable are not imported
+# exit with 0 otherwise
+$ cdk diff --retain AWS::DynamoDB::Table,Custom::DynamoDBReplica --import AWS::DynamoDB::GlobalTable
+
+Stack DemoStack
+✅ There were no differences
+```
+
+**Step 3:**: CDK users follow the construct upgrade guide to deploy the changes by running the
+CDK CLI command below:
+
+```s
+$ cdk deploy --import-existing-resources
```

Comment:
How about `cdk refactor --retain-delete-import`, which it definitely is going to have to be for this specific migration.

The workflow is then always 

1. Update code
2. `cdk diff --validate-whatever`
3. `cdk refactor`